### PR TITLE
Block header serialization

### PR DESCRIPTION
**Pull request by tomaszslabon**
This PR adds implementation of block header serialization function.

### CODE REVIEW

File: pkg/bitcoin/block_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	var expectedSerializedHeader [80]byte
+	copy(expectedSerializedHeader[:], expectedSerializedHeaderBytes)
+
+	if expectedSerializedHeader != actualSerializedHeader {
+		t.Errorf(
+			"\nexpected: %v\nactual:   %v",
+			expectedSerializedHeader,
+			actualSerializedHeader,
+		)
+	}
```
Comment (lukasz-zimnoch) [inline pkg/bitcoin/block_test.go:57]: Why not just:
```suggestion
	testutils.AssertBytesEqual(t, expectedSerializedHeader, actualSerializedHeader[:])
```

File: pkg/bitcoin/block_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		PreviousBlockHeaderHash: previousBlockHeaderHash,
+		MerkleRootHash:          merkleRootHash,
+		Time:                    1641914003,
+		Bits:                    436256810,
+		Nonce:                   778087099,
+	}
+
+	actualSerializedHeader := blockHeader.Serialize()
+
+	expectedSerializedHeaderBytes, err := hex.DecodeString(
```
Comment (lukasz-zimnoch) [inline pkg/bitcoin/block_test.go:39]:
```suggestion
	expectedSerializedHeader, err := hex.DecodeString(
```

Comment (tomaszslabon) [reply pkg/bitcoin/block_test.go]: Done in d389041340c2146f2640ff664a88a34594eb3751.

File: pkg/bitcoin/block_test.go (hunk 1 above)
Comment (tomaszslabon) [reply pkg/bitcoin/block_test.go]: Makes sense, done in d389041340c2146f2640ff664a88a34594eb3751.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**